src/green_recognition/include/CircleDetection.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Created By: Robyn Castro
+ * Created On: June 15, 2017
+ * Description: Determines whether or not green is seen on
+ *              the screen then publishes a recommended twist
+ *              message.
+ * Usage:
+ * Subscribes to: camera // TODO: Determine topic names
+ * Publishes to:
+ */
+
+#ifndef GREEN_RECOGNITION_H
+#define GREEN_RECOGNITION_H
+
+// OpenCV
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/opencv.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/objdetect/objdetect.hpp>
+
+// Image Conversion
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <image_transport/subscriber.h>
+
+// ROS
+#include <ros/ros.h>
+#include <ros/ros.h>
+#include <geometry_msgs/Twist.h>
+
+// STD
+#include <vector>
+#include <string>
+
+// Snowbots
+#include <sb_utils.h>
+
+using namespace cv;
+
+class GreenRecognition {
+
+public:
+
+    // Constructors used for testing
+    GreenRecognition(std::string &image_path);
+
+    GreenRecognition();
+
+    /**
+     * Constructor
+     */
+    GreenRecognition(int argc, char **argv, std::string node_name);
+
+    /**
+     * Counts the number of circles found in the image.
+     *
+     * @param image to be parsed
+     */
+    int countCircles(const Mat &filtered_image);
+
+private:
+
+    /**
+     * Callback for the filtered image
+     *
+     * @param address of filtered image matrix
```

Comment:
We can probably just simplify this to `filtered image`. It's pretty clear it's a pointer, and all images are really just some form of matrix anyhow